grapher/controls/CollapsibleList.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { ReactNode } from "react"
+import { observable, action, computed } from "mobx"
+import { observer } from "mobx-react"
+import { throttle } from "grapher/utils/Util"
+import { Tippy } from "grapher/chart/Tippy"
+
+interface ListChild {
+    index: number
+    child: ReactNode
+}
+
+/** A UI component inspired by the "Priority+ Navbar" or "Progressively Collapsing Navbar"*/
+@observer
+export class CollapsibleList extends React.Component {
+    private outerContainerRef: React.RefObject<
+        HTMLDivElement
+    > = React.createRef()
+    private moreButtonRef: React.RefObject<HTMLLIElement> = React.createRef()
+    private outerContainerWidth: number = 0
+    private moreButtonWidth: number = 0
+    private itemsWidths: number[] = []
+
+    @observable private visibleItems: ListChild[] = []
+    @observable private dropdownItems: ListChild[] = []
+
+    constructor(props: any) {
+        super(props)
+        this.visibleItems = this.children
+    }
+
+    @computed get children(): ListChild[] {
+        return (
+            React.Children.map(this.props.children, (child, i) => {
+                return {
+                    index: i,
+                    child,
+                }
+            }) ?? []
+        )
+    }
+
+    private updateOuterContainerWidth() {
+        this.outerContainerWidth =
+            this.outerContainerRef.current?.clientWidth ?? 0
+    }
+
+    private calculateItemWidths() {
+        this.outerContainerRef.current
+            ?.querySelectorAll(".list-item")
+            .forEach((item) => this.itemsWidths.push(item.clientWidth))
+    }
+
+    private get numItemsVisible() {
```

Comment:
I would make a tiny little method in Utils for this one (that way you can easily create some unit tests for it).

Reply:
Done!
cc854e775f0423834126721be7b46b4f4a5ca8dd